### PR TITLE
Piesno fix

### PR DESCRIPTION
**Pull request by samuelstjean**
That should hopefully bring the new PR with my latest fixes, rebased against upstream master.

### CODE REVIEW

File: dipy/reconst/tests/test_dti.py [hunk 1]
```diff
@@ -79,6 +79,15 @@ def test_tensor_model():
             npt.assert_almost_equal(dtifit.fa[0,0,0], dtifit_to_relative.fa,
                                     decimal=3)
 
+    # Check that it works on signal that has already been normalized to S0:
```
Comment (samuelstjean) [inline dipy/reconst/tests/test_dti.py]: Not sure wtf that is doing here... Is that supposed to happen?

Comment (arokem) [inline dipy/reconst/tests/test_dti.py]: Well, it's one of your commits:

https://github.com/samuelstjean/dipy/commit/996833c5ba2a1be4bef26b39f35d4e0e36d0efcb

You can interactively rebase that commit out and force push it again, to
eliminate it.

**Comment by samuelstjean** [comment]
That's weird, it fails in the test on import dipy.denoise.noise_estimate

======================================================================

ERROR: Failure: ImportError (cannot import name estimate_sigma)

----------------------------------------------------------------------

Traceback (most recent call last):

File "/home/travis/build/arokem/dipy/venv/lib/python2.6/site-packages/nose/loader.py", line 414, in loadTestsFromName

addr.filename, addr.module)

File "/home/travis/build/arokem/dipy/venv/lib/python2.6/site-packages/nose/importer.py", line 47, in importFromPath

return self.importFromDir(dir_path, fqname)

File "/home/travis/build/arokem/dipy/venv/lib/python2.6/site-packages/nose/importer.py", line 94, in importFromDir

mod = load_module(part_fqname, fh, filename, desc)

File "/home/travis/build/arokem/dipy/venv/lib/python2.6/site-packages/dipy/denoise/tests/test_noise_estimate.py", line 7, in <module>

from dipy.denoise.noise_estimate import _inv_nchi_cdf, piesno, estimate_sigma

ImportError: cannot import name estimate_sigma

but the function seems to still be there in the file.

**Comment by arokem** [comment]
Hmm. Mysterious. I've added back `estimate_sigma` on my branch. Let's see
how that works for us.

**Comment by arokem** [comment]
Fixed on my branch - I think that if you rebase on that one and force-push
your PR, it should be fixed on yours as well.

Let me know when you've done that, and I will take a look.

**Comment by samuelstjean** [comment]
Weird, when I pulled the branch on my home computer, it complained about the mysterious function havign reappeared and I didn't need to rebase stuff. That should be it now hopefully.

**Comment by arokem** [comment]
Hmm - something seems fishy here. Your last commit is ginormous, and includes everything on master since the time you made that branch. How about you start by rebasing that branch on master, and then rebase it on my branch? Also, there still seems to be something missing (see test error).

**Comment by samuelstjean** [comment]
Ok, now I hope this does it.

**Comment by samuelstjean** [comment]
Ok, something weird definitely happens with the buildbot, it complains on lines that do not even exist when I compare the code here with the error message.

======================================================================
ERROR: dipy.denoise.tests.test_noise_estimate.test_piesno
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/arokem/dipy/venv/lib/python3.3/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/travis/build/arokem/dipy/venv/lib/python3.3/site-packages/dipy/denoise/tests/test_noise_estimate.py", line 29, in test_piesno
    sigma = piesno(test_piesno_data, N=8, alpha=0.01, l=1, eps=1e-10, return_mask=False)
  File "/home/travis/build/arokem/dipy/venv/lib/python3.3/site-packages/dipy/denoise/noise_estimate.py", line 119, in piesno
    itermax=itermax, eps=eps)
  File "/home/travis/build/arokem/dipy/venv/lib/python3.3/site-packages/dipy/denoise/noise_estimate.py", line 220, in piesno_3D
    return_mask=return_mask)
NameError: global name 'return_mask' is not defined

and line 119 and 220 are not even those depicted in the error message, any idea why is that?

**Comment by arokem** [comment]
This should do it:
https://github.com/arokem/dipy/commit/d18b32329566e2cd5a9b59d4cebd52e25d95bff4

**Comment by arokem** [comment]
https://github.com/samuelstjean/dipy/pull/2

**Comment by samuelstjean** [comment]
please work now...

**Comment by samuelstjean** [comment]
wth, this seems to remove the example and only provide my fixes without all the good stuff from the other PRs.

**Comment by arokem** [comment]
You might want to close this, and start a new branch (off of my https://github.com/nipy/dipy/pull/576) to make these suggested changes.

**Comment by samuelstjean** [comment]
Well, last test before I give up this meta-rebasing which seems broken beyond repair. I manually merged mdesco/piesno_example this time, with everything hopefully pulled in.

**Comment by arokem** [comment]
Good luck!

**Comment by arokem** [comment]
Seems to have worked, in some sense, but ugliest commit log in the history of version control. I'll take a look at the code in detail

**Comment by samuelstjean** [comment]
yay... This was so beyond complicated for no apparent reason though, I think the original branch Maxime was working on and from which I added some stuff were never ever fully in sync, hence the problems.

**Comment by samuelstjean** [comment]
Ok, the example decided to disappear once again, so screw this and let's restart anew?

**Comment by arokem** [comment]
Yeah - please do that. Start with my branch, and make your changes on top
of that.

**Pull request closed without merging**